kernel/src/ipc/mod.rs

```diff
@@ -61,60 +145,160 @@ pub extern "C" fn syscall_entry(proto_method: u128, a: usize, b: usize, c: usize
 	}
 }
 
-fn syscall(proto_method: u128, a: usize, b: usize, _c: usize, _d: usize) -> Result<NonNegativeIsize, Error> {
-	if proto_method == 0 /* open@core.socket */ {
+fn syscall(proto_method: u128, a: usize, b: usize, c: usize, d: usize) -> Result<NonNegativeIsize, Error> {
+	if proto_method == 0 /* open@core.server.ctl */ {
```

Comment:
should say `open@core.object.Object`
```suggestion
	if proto_method == 0 /* open@core.object.Object */ {
```